xdsl/parser/attribute_parser.py

```diff
@@ -701,15 +778,33 @@ def _parse_builtin_dense_attr(self, _name: Span) -> DenseIntOrFPElementsAttr:
                 "Expected vector or tensor type of " "integer, index, or float type"
             )
 
+        # Get values and shape in case of hex_string (requires parsed type)
+        if hex_string is not None:
+            values, shape = self._parse_builtin_dense_attr_hex(hex_string, type)
```

Comment:
I might be missing something, but what changes if this line is inlined above, and the `hex_string` doesn't escape the `else:` branch above? I have a feeling that this method would be easier to understand with an `is_hex_string` `bool` value instead of the string, just to communicate that it's not used for anything once its parsed